File: clog-dart/lib/db/repo/repo.dart
/*
 * Copyright (c) 2022.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

library db_repo;

import 'package:clog/app/app.dart';
import 'package:clog/db/entity/dbq_log_entity.dart';
import 'package:clog/db/entity/info_log_entity.dart';
import 'package:clog/db/entity/service_log_entity.dart';
import 'package:fdation/fdation.dart';

part 'src/dbq_log_repo.dart';
part 'src/info_log_repo.dart';
part 'src/service_log_repo.dart';

final repo = _Repo();

class _Repo {
  late final infoLog = _InfoLogRepo();
  late final serviceLog = _ServiceLogRepo();
  late final dbqLog = _DbqLogRepo();
}

File: clog-go/db/repo/repo.go
/*
 * Copyright (c) 2023.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

package repo

var (
	DbqLog     *srDbqLog
	InfoLog    *srInfoLog
	ServiceLog *srServiceLog
)

func init() {
	DbqLog = srDbqLog{}.new()
	InfoLog = srInfoLog{}.new()
	ServiceLog = srServiceLog{}.new()
}
